sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -234,6 +239,24 @@ export class KeysClient {
     this.client = new KeyVaultClient(credential, SERVICE_API_VERSION, this.pipeline);
   }
 
+  private async deleteKey(name: string, options?: BeginDeleteKeyOptions): Promise<DeletedKey> {
+    const requestOptions = (options && options.requestOptions) || {};
```

Comment:
nit: usually prefer to default `options = {}` and have createSpan do the same so you can just pass options.requestOptions into createSpan without doing a bunch of conditional checks

Reply:
I understand, but since this is a copy of the original public deleteKey function, I don't think this is related to this PR. I think it would actually add a lot of noise to this PR.